tests fails under interactive shell (color issue)

Hey there,

Looks like with my regular shell setup the tests fails. I have the issue with abuild (Alpine Linux) or with regular make test. I guess as an interactive shell, some git commands output are colored (a magenta "John Doe").


```
=== RUN   TestCommentAdd
Building identity cache... Done.
Building bug cache... Done.
    comment_add_test.go:32: 
        	Error Trace:	/home/willow/sources/git-bug/commands/comment_test.go:151
        	           				/home/willow/sources/git-bug/commands/comment_add_test.go:32
        	Error:      	Not equal: 
        	           	expected: "John Doe"
        	           	actual  : "\x1b[35mJohn Doe\x1b[0m"
        	           	
        	           	Diff:
        	           	--- Expected
        	           	+++ Actual
        	           	@@ -1 +1 @@
        	           	-John Doe
        	           	+John Doe
        	Test:       	TestCommentAdd
--- FAIL: TestCommentAdd (0.02s)
=== RUN   TestCommentEdit
Building identity cache... Done.
Building bug cache... Done.
    comment_edit_test.go:17: 
        	Error Trace:	/home/willow/sources/git-bug/commands/comment_edit_test.go:17
        	Error:      	Received unexpected error:
        	           	comment doesn't exist
        	Test:       	TestCommentEdit
--- FAIL: TestCommentEdit (0.03s)
=== RUN   TestComment
Building identity cache... Done.
Building bug cache... Done.
    comment_test.go:20: 
        	Error Trace:	/home/willow/sources/git-bug/commands/comment_test.go:151
        	           				/home/willow/sources/git-bug/commands/comment_test.go:20
        	Error:      	Not equal: 
        	           	expected: "John Doe"
        	           	actual  : "\x1b[35mJohn Doe\x1b[0m"
        	           	
        	           	Diff:
        	           	--- Expected
        	           	+++ Actual
        	           	@@ -1 +1 @@
        	           	-John Doe
        	           	+John Doe
        	Test:       	TestComment
--- FAIL: TestComment (0.02s)
``` 

I successfully built and test with piping "make test | less".

--- a/commands/bug/testenv/testenv.go
+++ b/commands/bug/testenv/testenv.go
@@ -3,6 +3,7 @@ package testenv
 import (
 	"testing"
 
+	"github.com/fatih/color"
 	"github.com/stretchr/testify/require"
 
 	"github.com/MichaelMure/git-bug/commands/execenv"
@@ -16,6 +17,22 @@ const (
 
 func NewTestEnvAndUser(t *testing.T) (*execenv.Env, entity.Id) {
 	t.Helper()
+
+	// The Go testing framework either uses os.Stdout directly or a buffer
+	// depending on how the command is initially launched.  This results
+	// in os.Stdout.Fd() sometimes being a Terminal, and other times not
+	// being a Terminal which determines whether the ANSI library sends
+	// escape sequences to colorize the text.
+	//
+	// The line below disables all colorization during testing so that the
+	// git-bug command output is consistent in all test scenarios.
+	//
+	// See:
+	// - https://github.com/MichaelMure/git-bug/issues/926
+	// - https://github.com/golang/go/issues/57671
+	// - https://github.com/golang/go/blob/f721fa3be9bb52524f97b409606f9423437535e8/src/cmd/go/internal/test/test.go#L1180-L1208
+	// - https://github.com/golang/go/issues/34877
+	color.NoColor = true
 
 	testEnv := execenv.NewTestEnv(t)
 

